### DISCUSSION TITLE
geom.Contains expected behavior

### DISCUSSION THREAD

**Discussion by smallfish06** [peterstace/simplefeatures#639] [Q&A] [answered]
Thank you for this awesome project

In function description, `geom.Contains` allow boundary of A.
```go

// Contains returns true if and only if geometry A contains geometry B.
// Formally, the following two conditions must hold:
//
// 1. No points of B lies on the exterior of geometry A. That is, B must only
// be in the exterior or boundary of A.
//
// 2 .At least one point of the interior of B lies on the interior of A. That
// is, they can't *only* intersect at their boundaries.
func Contains(a, b Geometry) (bool, error) {
	return relateMatchesAnyPattern(a, b, "T*****FF*")
}
```

But while running below code, result is `false`.
```go
	a := geom.NewPolygon(
		[]geom.LineString{
			geom.NewLineString(geom.NewSequence([]float64{0, -10, 0, 10, 5, 10, 5, -10, 0, -10}, geom.DimXY)),
		}).AsGeometry()
	b := geom.NewPoint(geom.Coordinates{XY: geom.XY{X: 0, Y: 0}}).AsGeometry()
	
	contains, err := geom.Contains(a, b)
	if err != nil {
		return
	}
	fmt.Println(contains)
```

https://go.dev/play/p/JP_jmoRgylV

I'm sure I'm doing something wrong or misunderstand of description, what is `boundary` of point?

**Comment by peterstace** [comment, marked as answer]
Hi @smallfish06,

> what is boundary of point?

The boundary of a point is always the empty geometry.

> But while running below code, result is false.

The `Contains` predicate is false because the 2nd condition (below) is not met.

> At least one point of the interior of B lies on the interior of A. That is, they can't *only* intersect at their boundaries.

In particular, the interior of B (which is the point itself) _doesn't_ lie on the interior of A (the rectangle). Instead, it lies on the boundary of A.

---

Hopefully that helps!

**Reply by smallfish06** [reply]
Thank you for detailed answer! Have a great weekend 😄
